sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/SaplingParamTool.kt

```diff
@@ -17,128 +17,141 @@ import java.nio.channels.Channels
 // TODO [#665]: https://github.com/zcash/zcash-android-wallet-sdk/issues/665
 // TODO [#665]: Recover from corrupted sapling-spend.params and sapling-output.params
 
-// TODO [#664]: https://github.com/zcash/zcash-android-wallet-sdk/issues/664
-// TODO [#664]: Check size of download for sapling-spend.params and sapling-output.params
-
 // TODO [#611]: https://github.com/zcash/zcash-android-wallet-sdk/issues/611
 // TODO [#611]: Move Params Directory to No Backup Directory
 
-@Suppress("UtilityClassWithPublicConstructor")
-class SaplingParamTool {
+object SaplingParamTool {
+    /**
+     * Maximum file size for the sapling spend params - 50MB
+     */
+    internal const val SPEND_PARAM_FILE_MAX_BYTES_SIZE = 50L * 1024L * 1024L
 
-    companion object {
-        /**
-         * Checks the given directory for the output and spending params and calls [fetchParams] if
-         * they're missing.
-         *
-         * @param destinationDir the directory where the params should be stored.
-         */
-        suspend fun ensureParams(destinationDir: String) {
-            var hadError = false
-            arrayOf(
-                ZcashSdk.SPEND_PARAM_FILE_NAME,
-                ZcashSdk.OUTPUT_PARAM_FILE_NAME
-            ).forEach { paramFileName ->
-                if (!File(destinationDir, paramFileName).existsSuspend()) {
-                    twig("WARNING: $paramFileName not found at location: $destinationDir")
-                    hadError = true
-                }
+    /**
+     * Maximum file size for the sapling spend params - 5MB
+     */
+    internal const val OUTPUT_PARAM_FILE_MAX_BYTES_SIZE = 5L * 1024L * 1024L
+
+    /**
+     * Checks the given directory for the output and spending params and calls [fetchParams] if
+     * they're missing.
+     *
+     * @param destinationDir the directory where the params should be stored.
+     */
+    suspend fun ensureParams(destinationDir: String) {
+        var hadError = false
+        arrayOf(
+            ZcashSdk.SPEND_PARAM_FILE_NAME,
+            ZcashSdk.OUTPUT_PARAM_FILE_NAME
+        ).forEach { paramFileName ->
+            if (!File(destinationDir, paramFileName).existsSuspend()) {
+                twig("WARNING: $paramFileName not found at location: $destinationDir")
+                hadError = true
             }
-            if (hadError) {
-                @Suppress("TooGenericExceptionCaught")
-                try {
-                    Bush.trunk.twigTask("attempting to download missing params") {
-                        fetchParams(destinationDir)
-                    }
-                } catch (e: Throwable) {
-                    twig("failed to fetch params due to: $e")
-                    throw TransactionEncoderException.MissingParamsException
+        }
+        if (hadError) {
```

Comment:
I'm realizing that we have a potential future optimization here.  We should be doing a arrayOf().filter(it.exists()).forEach {download()}

Right now, if one file is missing then we redownload both.

Reply:
I've decided to implement this requirement too, so we can build on it in the subsequent tasks. Please, check its new behaviour. Only the `SaplingParamTool` and its tests were changed.